lib/page/needed.rb

```diff
@@ -0,0 +1,38 @@
+require 'octokit'
+
+module Page
+  class Needed
+    def initialize(access_token: required)
+      token_error if access_token.to_s.strip.empty?
+      @access_token = access_token
+    end
+
+    def required
+      token_error
```

Comment:
This is an interesting approach, but I'm not sure what you gain from the indirection of a `required`  method _and_ a `token_error` method, rather than just one or the other.